test/distributed/pipelining/test_schedule.py

```diff
@@ -52,6 +52,46 @@ def setUpClass(cls):
         dev_id = cls.rank % torch.cuda.device_count()
         cls.device = torch.device(f"cuda:{dev_id}")
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    @parametrize("ScheduleClass", [ScheduleGPipe, Schedule1F1B])
+    def test_multi_iter(self, ScheduleClass):
+        mod = MultiMLP(d_hid, n_layers=self.world_size)
+        mod.to(self.device)
+
+        x = torch.randn(batch_size, d_hid, device=self.device)
+        target = torch.randn(batch_size, d_hid, device=self.device)
+        loss_fn = torch.nn.MSELoss(reduction="sum")
+
+        # Create a pipeline
+        chunks = 4
+        split_spec = mod.split_spec if hasattr(mod, "split_spec") else None
+        pipe = pipeline(
+            mod,
+            chunks,
+            example_args=(x,),
+            split_spec=split_spec,
+        )
+
+        stage = PipelineStage(
+            pipe,
+            self.rank,
+            device=self.device,
+        )
+
+        # Attach to a schedule
+        schedule = ScheduleClass(stage, chunks, loss_fn=loss_fn)
+
+        # Run
+        for _ in range(20):
```

Comment:
nit: would it make more sense to just modify the gradient tests to run multiple iterations, and then the multi-iter test is also testing correctness?

Reply:
I gave it more thoughts but still prefer for these two tests to indicate two different failures. 
This stress test is not for checking numerics, but to catch if our schedules would hang after many iterations.

If any consolidation can be made, I will be happy to make it. At this point I just want to land more tests before release to safeguard library changes.